From 4858faa662f25c4e1e6bb872b4b1d5daf92c790b Mon Sep 17 00:00:00 2001
From: Chris Zhong <zyw@rock-chips.com>
Date: Thu, 9 Nov 2017 11:36:35 +0800
Subject: [PATCH] rk_nand: change the nandc clk to optional

RK3066 NANDC has not clk, there is only a hclk, the working frequency of
nandc is same as this hclk, so change it to optional.

Change-Id: I638c26f33a3ff9eaa5c532168e05590ed048bcf9
Signed-off-by: Chris Zhong <zyw@rock-chips.com>
---
 drivers/rk_nand/rk_nand_base.c | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/drivers/rk_nand/rk_nand_base.c b/drivers/rk_nand/rk_nand_base.c
index 77fe609484a6..548ea61987bd 100644
--- a/drivers/rk_nand/rk_nand_base.c
+++ b/drivers/rk_nand/rk_nand_base.c
@@ -379,22 +379,24 @@ static int rknand_probe(struct platform_device *pdev)
 	g_nandc_info[id].clk = devm_clk_get(&pdev->dev, "clk_nandc");
 	g_nandc_info[id].gclk = devm_clk_get(&pdev->dev, "g_clk_nandc");
 
-	if (unlikely(IS_ERR(g_nandc_info[id].clk)) ||
-	    unlikely(IS_ERR(g_nandc_info[id].hclk))) {
-		dev_err(&pdev->dev, "rknand_probe get clk error\n");
-		return -1;
+	if (unlikely(IS_ERR(g_nandc_info[id].hclk))) {
+		dev_err(&pdev->dev, "rknand_probe get hclk error\n");
+		return PTR_ERR(g_nandc_info[id].hclk);
+	}
+
+	if (!(IS_ERR(g_nandc_info[id].clk))) {
+		clk_set_rate(g_nandc_info[id].clk, 150 * 1000 * 1000);
+		g_nandc_info[id].clk_rate = clk_get_rate(g_nandc_info[id].clk);
+		clk_prepare_enable(g_nandc_info[id].clk);
+		dev_info(&pdev->dev,
+			 "rknand_probe clk rate = %d\n",
+			 g_nandc_info[id].clk_rate);
 	}
 
-	clk_set_rate(g_nandc_info[id].clk, 150 * 1000 * 1000);
-	g_nandc_info[id].clk_rate = clk_get_rate(g_nandc_info[id].clk);
-	clk_prepare_enable(g_nandc_info[id].clk);
 	clk_prepare_enable(g_nandc_info[id].hclk);
 	if (!(IS_ERR(g_nandc_info[id].gclk)))
 		clk_prepare_enable(g_nandc_info[id].gclk);
 
-	dev_info(&pdev->dev,
-		 "rknand_probe clk rate = %d\n",
-		 g_nandc_info[id].clk_rate);
 	return 0;
 }
 
-- 
2.35.3

